Doc/library/random.rst

```diff
@@ -516,6 +516,42 @@ Simulation of arrival times and service deliveries for a multiserver queue::
     print(f'Mean wait: {mean(waits):.1f}   Max wait: {max(waits):.1f}')
     print('Quartiles:', [round(q, 1) for q in quantiles(waits)])
 
+Recipes
+-------
+
+The default :func:`.random` returns one of 2⁵³ evenly spaced floats in
+the range ``0.0 ≤ x < 1.0``.  Each possible selection is binary fraction
+with an integer numerator in the range ``0 ≤ X < 2⁵³`` and with a
+denominator of 2⁵³. A number like ``0.05954861408025609`` is not a
+possible selection because its binary fraction, ``4290929858916333 ÷ 2⁵⁶``
+has a denominator that is greater than 2⁵³.
+
+For some applications, it may be desirable to have an alternative
+implementation of :func:`.random` that selects from a much larger
+population that includes binary fractions with larger denominators. The
+following recipe chooses floats in the range ``0.0 ≤ x < 1.0`` but
+allows denominators in the full range supported by Python floats::
+
+    from random import getrandbits
+
+    def full_random():
+        ''' Uniform distribution from all possible floats
+            in the interval 0.0 <= X < 1.0.
+
+        '''
+        # Choose a significand uniformly from:  2**52 <= mantissa < 2**53.
+        # Then choose a binade from a geometric distribution where smaller
+        # binade exponents occur half as often as the next larger exponent.
+
+        mantissa = 0x10_0000_0000_0000 | getrandbits(52)
+        exponent = -53
+        x = 0
+        while not x:
+            x = getrandbits(32)
+            exponent += x.bit_length() - 32
+        return mantissa * 2.0 ** exponent
```

Comment:
Multiplication rounds, so when this slobbers into the denorm range, nearest/even rounding will give a slight bias toward 0 in the last retained bit.  `ldexp()` on Windows truncates instead, which doesn't introduce bias in the denorm cases; but I believe `ldexp()` on most other platforms does round.

Reply:
+1 for this use of "slobber"